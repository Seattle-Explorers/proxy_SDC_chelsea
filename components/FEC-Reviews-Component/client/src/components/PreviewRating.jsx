import React from 'react';
import styled from 'styled-components';

const RatingPreviewContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(2, auto);
`

const CategoryContainer = styled.div`
  width: 100%;
  display: flex;
  justify-content: space-between;
  margin-left: 50px;
  margin-top: 7px;
  margin-bottom: 7px;
`

const Bar = styled.div`
  width: ${props => props.width};
  height: ${props => props.height};
  float: left;
  border-radius: 15%;
  background-color: ${props => props.color};
  margin-right: 5px;
`
const TextContainer = styled.div`
  width: 25%;
  flex-direction: row;
`
const Rating = styled.span`
  font-size: 15px;
  font-weight: bolder;
`

const RatingContainer = styled.div`
  display: flex;
  justify-content: flex-end;
  height: 100%;
  width: 75%;
  margin-right: 10px;
`
const PreviewRating = ({ overview }) => {

  const { cleanAvg, commAvg, accuracyAvg, valueAvg, locationAvg, checkinAvg } = overview;
  const cleanPercent = Math.floor((Number(cleanAvg) / 5) * 100);
  const commPercent = Math.floor((Number(commAvg) / 5) * 100);
  const accuracyPercent = Math.floor((Number(accuracyAvg) / 5) * 100);
  const valuePercent = Math.floor((Number(valueAvg) / 5) * 100);
  const locationPercent = Math.floor((Number(locationAvg) / 5) * 100);
  const checkinPercent = Math.floor((Number(checkinAvg) / 5) * 100);

  return (
    <RatingPreviewContainer>
      <CategoryContainer>
        <TextContainer>Cleanliness</TextContainer>
        <RatingContainer>
          <Bar width='33%' color='#B3B3B3' height='0.5em'>
            <Bar width={`${cleanPercent}%`} color='black' height='100%'/> 
          </Bar>
          <Rating>{commAvg}</Rating>
        </RatingContainer>
      </CategoryContainer>

      <CategoryContainer>
        <TextContainer>Accuracy</TextContainer>
        <RatingContainer>
          <Bar width='33%' color='#B3B3B3' height='0.5em'>
            <Bar width={`${accuracyPercent}%`} color='black' height='100%'/>
          </Bar> 
          <Rating>{accuracyAvg}</Rating>
        </RatingContainer>
      </CategoryContainer>

      <CategoryContainer>
        <TextContainer>Communication</TextContainer>
        <RatingContainer>
          <Bar width='33%' color='#B3B3B3' height='0.5em'>
            <Bar width={`${commPercent}%`} color='black' height='100%'/> 
          </Bar>
          <Rating>{commAvg}</Rating>
        </RatingContainer>
      </CategoryContainer>

      <CategoryContainer>
        <TextContainer>Location</TextContainer>
        <RatingContainer>
          <Bar width='33%' color='#B3B3B3' height='0.5em'>
            <Bar width={`${locationPercent}%`} color='black' height='100%'/>
          </Bar> 
          <Rating>{locationAvg}</Rating>
        </RatingContainer>
      </CategoryContainer>

      <CategoryContainer>
        <TextContainer>Check-in</TextContainer>
        <RatingContainer>
          <Bar width='33%' color='#B3B3B3' height='0.5em'>
            <Bar width={`${checkinPercent}%`} color='black' height='100%'/> 
          </Bar>
          <Rating>{checkinAvg}</Rating>
        </RatingContainer>
      </CategoryContainer>

      <CategoryContainer>
        <TextContainer>Communication</TextContainer>
        <RatingContainer>
          <Bar width='33%' color='#B3B3B3' height='0.5em'>
            <Bar width={`${valuePercent}%`} color='black' height='100%'/>
          </Bar> 
          <Rating>{valueAvg}</Rating>
        </RatingContainer>
      </CategoryContainer>

    </RatingPreviewContainer>
  )
}

export default PreviewRating;
