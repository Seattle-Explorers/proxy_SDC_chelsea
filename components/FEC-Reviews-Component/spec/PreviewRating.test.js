import 'jsdom-global/register';
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-enzyme';
import 'jest-styled-components';

import {
  cleanAvg, commAvg, accuracyAvg, valueAvg, locationAvg, checkinAvg,
} from '../__mocks__/mockData';
import PreviewRating from '../client/src/components/PreviewRating';

describe('Preview Rating Components', () => {
  const stats = {
    cleanAvg,
    commAge: commAvg,
    accuracyAvg,
    valueAvg,
    locationAvg,
    checkinAvg,
  };

  it('should render', () => {
    const component = <PreviewRating overview={stats} />;

    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
