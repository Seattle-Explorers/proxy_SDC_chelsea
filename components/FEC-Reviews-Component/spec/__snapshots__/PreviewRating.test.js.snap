// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview Rating Components should render 1`] = `
.c0 {
  display: grid;
  grid-template-columns: repeat(2,auto);
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-left: 50px;
  margin-top: 7px;
  margin-bottom: 7px;
}

.c4 {
  width: 33%;
  height: 0.5em;
  float: left;
  border-radius: 15%;
  background-color: #B3B3B3;
  margin-right: 5px;
}

.c5 {
  width: 20%;
  height: 100%;
  float: left;
  border-radius: 15%;
  background-color: black;
  margin-right: 5px;
}

.c7 {
  width: 24%;
  height: 100%;
  float: left;
  border-radius: 15%;
  background-color: black;
  margin-right: 5px;
}

.c8 {
  width: NaN%;
  height: 100%;
  float: left;
  border-radius: 15%;
  background-color: black;
  margin-right: 5px;
}

.c9 {
  width: 80%;
  height: 100%;
  float: left;
  border-radius: 15%;
  background-color: black;
  margin-right: 5px;
}

.c10 {
  width: 38%;
  height: 100%;
  float: left;
  border-radius: 15%;
  background-color: black;
  margin-right: 5px;
}

.c11 {
  width: 32%;
  height: 100%;
  float: left;
  border-radius: 15%;
  background-color: black;
  margin-right: 5px;
}

.c2 {
  width: 25%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c6 {
  font-size: 15px;
  font-weight: bolder;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  height: 100%;
  width: 75%;
  margin-right: 10px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      Cleanliness
    </div>
    <div
      className="c3"
    >
      <div
        className="c4"
        color="#B3B3B3"
        height="0.5em"
        width="33%"
      >
        <div
          className="c5"
          color="black"
          height="100%"
          width="20%"
        />
      </div>
      <span
        className="c6"
      />
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      Accuracy
    </div>
    <div
      className="c3"
    >
      <div
        className="c4"
        color="#B3B3B3"
        height="0.5em"
        width="33%"
      >
        <div
          className="c7"
          color="black"
          height="100%"
          width="24%"
        />
      </div>
      <span
        className="c6"
      >
        1.2
      </span>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      Communication
    </div>
    <div
      className="c3"
    >
      <div
        className="c4"
        color="#B3B3B3"
        height="0.5em"
        width="33%"
      >
        <div
          className="c8"
          color="black"
          height="100%"
          width="NaN%"
        />
      </div>
      <span
        className="c6"
      />
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      Location
    </div>
    <div
      className="c3"
    >
      <div
        className="c4"
        color="#B3B3B3"
        height="0.5em"
        width="33%"
      >
        <div
          className="c9"
          color="black"
          height="100%"
          width="80%"
        />
      </div>
      <span
        className="c6"
      >
        4
      </span>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      Check-in
    </div>
    <div
      className="c3"
    >
      <div
        className="c4"
        color="#B3B3B3"
        height="0.5em"
        width="33%"
      >
        <div
          className="c10"
          color="black"
          height="100%"
          width="38%"
        />
      </div>
      <span
        className="c6"
      >
        1.9
      </span>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      Communication
    </div>
    <div
      className="c3"
    >
      <div
        className="c4"
        color="#B3B3B3"
        height="0.5em"
        width="33%"
      >
        <div
          className="c11"
          color="black"
          height="100%"
          width="32%"
        />
      </div>
      <span
        className="c6"
      >
        1.6
      </span>
    </div>
  </div>
</div>
`;
